controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -300,6 +301,16 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (res ctrl.Result, reterr error) {
 	rosaScope.Info("Reconciling ROSAControlPlane delete")
 
+	// Deleting MachinePools first.
+	deleted, err := r.deleteMachinePools(ctx, rosaScope)
```

Comment:
Forgive my ignorance of ROSA here, but by comparison, the EKS control plane controller does not delete machine pools.

AFAIK, either the user deletes them directly, or relies on the topology controller (for a ClusterClass cluster) to do so.

Reply:
Not sure about EKS , but from what I can see eks controlPlane is creating/deleting the infra (network, subnets,...etc) while rosa controlPlane creates/deletes the rosaCluster and RosaMachinePool. 
The PR is fixing this [issue-4936](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4936) which as explained handling the gitops workflow for deprovision the clusters. It has a related [issue-10544](https://github.com/kubernetes-sigs/cluster-api/issues/10544) for the cluster-api machinePool as well.